ansq/tcp/types/message.py

```diff
@@ -9,51 +9,50 @@
 
     from . import NSQMessageSchema
 
-__all__ = "NSQMessage"
+__all__ = ["NSQMessage"]
 
 
-def not_processed(func: Callable) -> Callable:
-    """Decorator to verify that the message has not yet been processed.
+def ensure_can_be_processed(func: Callable) -> Callable:
+    """Decorator to verify that the message can be processed.
 
     :raises RuntimeWarning: in case message was processed earlier.
     """
 
     @wraps(func)
-    async def decorator(cls: "NSQMessage", *args: Any, **kwargs: Any) -> Any:
-        if cls.is_processed:
-            raise RuntimeWarning("Message has already been processed")
-        response = await func(cls, *args, **kwargs)
-        return response
+    async def wrapper(message: "NSQMessage", *args: Any, **kwargs: Any) -> Any:
+        if message.is_processed:
+            raise RuntimeWarning(f"Message id={message.id} has already been processed")
```

Comment:
While `RuntimeWarning`  is technically an exception, it should not be raised manually in code, because conceptually it belongs to the [warnings mechanism](https://docs.python.org/3/library/warnings.html#warning-categories). This must be fixed in a following-up PR.

Reply:
Tracking issue - https://github.com/list-family/ansq/issues/72.